core/market/decentralized/src/db/models/agreement.rs

```diff
@@ -0,0 +1,121 @@
+use chrono::{NaiveDateTime, Utc};
+use diesel::backend::Backend;
+use diesel::deserialize;
+use diesel::serialize::Output;
+use diesel::sql_types::Integer;
+use diesel::types::{FromSql, ToSql};
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
+
+use ya_client::model::NodeId;
+
+use crate::db::models::{OwnerType, Proposal};
+use crate::db::schema::market_agreement;
+use crate::ProposalId;
+
+pub type AgreementId = ProposalId;
+
+/// TODO: Could we avoid having separate enum type for database
+///  and separate for client?
+#[derive(FromPrimitive, AsExpression, FromSqlRow, PartialEq, Debug, Clone, Copy)]
+#[sql_type = "Integer"]
+pub enum AgreementState {
+    /// Newly created by a Requestor (based on Proposal)
+    Proposal = 0,
+    /// Confirmed by a Requestor and send to Provider for approval
+    Pending = 1,
+    /// Cancelled by a Requestor
+    Cancelled = 2,
+    /// Rejected by a Provider
+    Rejected = 3,
+    /// Approved by both sides
+    Approved = 4,
+    /// Not accepted, rejected nor cancelled within validity period
+    Expired = 5,
+    /// Finished after approval
+    Terminated = 6,
+}
+
+#[derive(Clone, Debug, Identifiable, Insertable, Queryable)]
+#[table_name = "market_agreement"]
+pub struct Agreement {
+    pub id: AgreementId,
+
+    pub offer_properties: String,
+    pub offer_constraints: String,
+
+    pub demand_properties: String,
+    pub demand_constraints: String,
+
+    pub provider_id: NodeId,
+    pub requestor_id: NodeId,
+
+    /// End of validity period.
+    /// Agreement needs to be accepted, rejected or cancelled before this date; otherwise will expire.
+    pub creation_ts: NaiveDateTime,
```

Comment:
`ts` suffix is misleading for me. It's a datetime not a float/int.

Reply:
Make a new issue, and leave in this PR as is.

Reply:
To be compatible with rest of yagna modules we decided to use `_date` suffix

Reply:
#416

Reply:
ok